### PR TITLE
[SP-5688] Add abstract cloud storage

### PR DESCRIPTION
**Pull request by moshebe**

### CODE REVIEW

File: storage/storage.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/stackpulse/steps-sdk-go/cloud/gcp"
+	"github.com/stackpulse/steps-sdk-go/log"
+	"gocloud.dev/blob"
+
+	_ "gocloud.dev/blob/gcsblob"
+	_ "gocloud.dev/blob/s3blob"
```
Comment (kostyay) [inline storage/storage.go:14]: is this needed?

File: storage/storage.go [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+		if _, err := gcp.SetAppCredentials(); err != nil {
+			return fmt.Errorf("set gcp credentials: %w", err)
+		}
+	}
+
+	return nil
+}
+
+// UploadFile to the given bucket by where file is the local path and key is the desired remote path in the bucket.
+func (s *Storage) UploadFile(ctx context.Context, bucket *blob.Bucket, file, key string) error {
```
Comment (kostyay) [inline storage/storage.go:41]:
```suggestion
func (s *Storage) Upload(ctx context.Context, file string, bucket *blob.Bucket, key string) error {
```
i think it makes more sense to pass file param before bucket as its upload from -> to so it should come first
also id call this Upload and the other function UploadFIles

File: storage/storage.go (hunk 1 above)
Comment (moshebe) [reply storage/storage.go:14]: Yes, it's a go cloud requirement

File: storage/storage.go [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+	_, err = io.Copy(w, f)
+	if err != nil {
+		return fmt.Errorf("copy: %w", err)
+	}
+
+	return nil
+}
+
+// Upload puts the wanted files given by the map between local and remote paths into the bucket that was defined by `Bucket`.
+func (s *Storage) Upload(ctx context.Context, files map[string]string) error {
```
Comment (kostyay) [inline storage/storage.go:63]:
```suggestion
func (s *Storage) UploadFiles(ctx context.Context, files map[string]string) error {
```
im not sure id use map[string]string here.
what if you want to upload the same file to multiple keys?
maybe an array is more fitting here
[]struct { src, key string }?

**Review by kostyay — APPROVED** [review]
LGTM

**Pull request merged**